StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/CustomSheet/CustomSheetViewController.swift

```diff
@@ -0,0 +1,206 @@
+//
+//  CustomSheetViewController.swift
+//  StripeFinancialConnections
+//
+//  Created by Krisjanis Gaidis on 6/2/23.
+//
+
+import Foundation
+import UIKit
+
+final class CustomSheetCommunicationHelper {
```

Comment:
This is a simple object that facilitates communication between the sheet infra and the view/viewcontroller that manages the sheet. 

It provides a `dismissSheet` function so the "sheet view" can decide to dismiss the sheet programatically.

---

Why design it like this? 

Usually the way the sheet setup works is:
0. ???
1. Create a view for the sheet
2. Present the view as a sheet

The problem is that (1) somehow needs to know about (2) [so (1) can dismiss (2)]. So we create `CustomSheetCommunicationHelper` as part of step (0), and then pass it to both (1) and (2). 

---

Note that I may be able to delete this, but am still debating it (will save for a different PR). Technically, if every sheet is presented from a UIViewController, then calling `self?.dismiss(animated: true)` from the SheetViewController would do the same thing. The problem is may not necessarily be desirable for every sheet to be a UIViewController, but it could at least simplify this + the interface so maybe its worth it.